### PR TITLE
ui changes

### PR DESCRIPTION
**Pull request by awildplace**
missing some new icons, as i couldn't figure out how to get them into github.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/hearthisandroid/1)

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
     android:versionName="1.0" >
 
     <uses-sdk
-        android:minSdkVersion="10"
+        android:minSdkVersion="11"
```
Comment (JohnThomson) [inline app/src/main/AndroidManifest.xml]: Since we're not very interested in tablets, this cuts out all devices before Android 4.0, I believe. What are we using that requires that?

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -30,16 +30,16 @@
...
         </activity>
         <activity
-            android:name=".ChooseChapterActivity"
-            android:label="@string/app_name" >
+            android:name=".ChaptersActivity"
+            android:label="" >
         </activity>
         <activity
             android:name=".RecordActivity"
-            android:label="@string/app_name" >
```
Comment (JohnThomson) [inline app/src/main/AndroidManifest.xml]: I'd like to understand why this should be taken out

File: app/src/main/java/Script/BookInfo.java [hunk 3]
```diff
@@ -1,22 +1,27 @@
...
 
-public class BookInfo implements Serializable {
-	private String _projectName;
-	public String Name;
-	public String Abbr;
-	public int ChapterCount;
-	public int BookNumber;
+public class BookInfo implements Serializable
+{
+	private String projectName;
```
Comment (JohnThomson) [inline app/src/main/java/Script/BookInfo.java]: This one is a tough call. I personally really like an obvious distinction between class member and local variables, and a leading underscore is the current convention in SIL code (mostly C#). But I see that Google at least does not do this for Java (https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s5.1-identifier-names). So I suppose you are right.

File: app/src/main/java/Script/BookInfo.java [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
+	private String name;
+	private String abbr;
+	private int chapterCount;
+	private int bookNumber;
 
-	public IScriptProvider getScriptProvider() {
+	public IScriptProvider getScriptProvider()
+	{
```
Comment (JohnThomson) [inline app/src/main/java/Script/BookInfo.java]: OTOH this is a C# standard which defies the google one we normally use for Java and JavaScript. https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.1-braces. Let's stick with no line break before the opening brace.

File: app/src/main/java/Script/BookInfo.java [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
-public class BookInfo implements Serializable {
-	private String _projectName;
-	public String Name;
-	public String Abbr;
-	public int ChapterCount;
-	public int BookNumber;
+public class BookInfo implements Serializable
+{
+	private String projectName;
+	private String name;
```
Comment (JohnThomson) [inline app/src/main/java/Script/BookInfo.java]: You're right about these...capitalizing public properties is a C# convention.

File: app/src/main/java/Script/IScriptProvider.java [hunk 6]
```diff
@@ -1,6 +1,9 @@
 package Script;
 
-public interface IScriptProvider {
+import org.json.JSONObject;
+
+public interface IScriptProvider
+{
```
Comment (JohnThomson) [inline app/src/main/java/Script/IScriptProvider.java]: I won't keep commenting on these, but please remove all similar changes. (I'm open to debating this, by the way, but we need to settle it before I merge this change set.)

File: app/src/main/java/Script/RealScriptProvider.java [hunk 7]
```diff
@@ -22,59 +20,73 @@
...
 import javax.xml.parsers.ParserConfigurationException;
 
-public class RealScriptProvider implements IScriptProvider {
-	
-	String _path;
-	List<BookData> Books = new ArrayList<BookData>();
+public class RealScriptProvider implements IScriptProvider
+{
+	String path;
+	ArrayList<BookData> books = new ArrayList<BookData>();
```
Comment (JohnThomson) [inline app/src/main/java/Script/RealScriptProvider.java]: Is there a particular reason to make the declaration more specific? See e.g. http://stackoverflow.com/questions/2279030/type-list-vs-type-arraylist-in-java

File: app/src/main/java/org/sil/hearthis/ChaptersActivity.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.GridView;
+import android.widget.TextView;
+
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+public class ChaptersActivity extends Activity
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/ChaptersActivity.java]: In a shared project, renaming files is a good thing to discuss and coordinate. As you can see, github treats it as adding and deleting the whole file, which makes it hard to identify and review any other changes. So it's best done as an isolated PR where there ARE no other changes (except the required ones where the renamed class is used.)

File: app/src/main/java/org/sil/hearthis/MainActivity.java [hunk 9]
```diff
@@ -1,76 +1,220 @@
...
-		// Todo: remember last project
-		// Todo: if no real project available use SampleScriptProvider.
-		RealScriptProvider scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+
+		IScriptProvider scripture;
+		try
+		{
+			scripture = new RealScriptProvider(getExternalFilesDir(null) + "/" + "Dhh");
+		}
+		catch (Exception ex)
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/MainActivity.java]: In general, I consider it bad practice to catch all exceptions like this. For example, if I introduce some coding bug into the RealScriptProvider constructor and generate a null reference exception, I don't want it ignored. Could you catch more specific exceptions that really make this response appropriate, like FileNotFound, or maybe IOException?

File: app/src/main/java/org/sil/hearthis/MainActivity.java [hunk 10]
```diff
@@ -1,76 +1,220 @@
...
+	@Override
+	public boolean onCreateOptionsMenu(Menu menu)
+	{
+		mainMenu = menu;
+		MenuInflater inflater = getMenuInflater();
+		inflater.inflate(R.menu.menu_main, menu);
+		return super.onCreateOptionsMenu(menu);
+	}
+
+	public void setProject(Project project)
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/MainActivity.java]: Again, moving existing methods makes it hard to see what else you changed. If you want to re-arrange a file, it's best to make a separate PR.

File: app/src/main/java/org/sil/hearthis/MainActivity.java [hunk 11]
```diff
@@ -1,76 +1,220 @@
...
+					myInfo.getAbbr() :
+					myInfo.getName().substring(0, 3));
+			return button;
+		}
+
+		private int setButtonColour(int bookNumber)
+		{
+			int navbuttoncolour = R.color.navButtonColor;
+			if (bookNumber < 5)
+			{
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/MainActivity.java]: Assuming we agree on the Google coding standard, I'd naturally like you to use it for your additions, too.

File: app/src/main/java/org/sil/hearthis/RecordActivity.java [hunk 12]
```diff
@@ -34,42 +33,52 @@
...
+	protected void onCreate(Bundle state)
+	{
+		super.onCreate(state);
 		setContentView(R.layout.activity_record);
 		
 		mtfl = (Typeface)Graphite.addFontResource(getAssets(), "CharisSILAfr-R.ttf", "charis", 0, "", "");
-		
+
+		if (state != null)
+			activeLine = state.getInt("activeLine");
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/RecordActivity.java]: It's a good idea to keep track of the active line. Note that we will need to reset it when advancing to the next chapter.

File: app/src/main/java/org/sil/hearthis/RecordActivity.java [hunk 13]
```diff
@@ -77,63 +86,70 @@ protected void onCreate(Bundle savedInstanceState) {
...
+			if (i == activeLine)
+			{
 				lineView.setTextColor(getResources().getColor(R.color.activeTextLine));
 			}
+			lineView.setOnClickListener(new OnClickListener()
+			{
+				@Override
+				public void onClick(View view)
+				{
+					((TextView)linesView.getChildAt(activeLine)).setTextColor(getResources().getColor(R.color.labelTextColor));
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/RecordActivity.java]: We're starting to have a few places that know how the active line is highlighted. I'd like to see a method for setting the active line that encapsulates all the code for updating the display (probably something I should have done myself, sorry.)

File: app/src/main/java/org/sil/hearthis/RecordActivity.java [hunk 14]
```diff
@@ -160,27 +179,74 @@ void nextButtonClicked() {
...
+	public boolean onKeyLongPress(int keyCode, KeyEvent event)
+	{
+		switch (keyCode)
+		{
+		case KeyEvent.KEYCODE_VOLUME_DOWN:
+		case KeyEvent.KEYCODE_VOLUME_UP:
+			// ignore the long press, as the button is being held
+			return true;
+		default:
+			return super.onKeyUp(keyCode, event);
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/RecordActivity.java]: It's odd to call the super version of a different method. If you really meant to do that, I think a comment indicating why is called for.

File: app/src/main/java/org/sil/hearthis/SyncServer.java [hunk 15]
```diff
@@ -89,16 +95,17 @@ public void run() {
                     httpService.handleRequest(serverConnection, httpContext);
 
                     serverConnection.shutdown();
-                } catch (IOException e) {
-                    e.printStackTrace();
-                } catch (HttpException e) {
+                }
+				catch (Exception e)
```
Comment (JohnThomson) [inline app/src/main/java/org/sil/hearthis/SyncServer.java]: See comment above on why I don't like a blanket catch of all exceptions.

**Comment by andrew-polk** [comment]
Given that this PR is 7.5 years old, it seems safe to close...
